arelle/Cntlr.py

```diff
@@ -619,6 +619,23 @@ def memoryUsed(self) -> float | int:
             pass
         return 0
 
+    def workingOnlineOrInCache(self, url: str) -> bool:
+        """
+        Determines if the given URL should be requested based on the web cache's internet connectivity status
+        and whether the URL already exists in the cache.
+        :param url: Web URL
+        :return: True if the URL should be requested, False if not
+        """
+        if not self.webCache.workOffline:
```

Comment:
At the moment this doesn't behave as expected from the command line. The work offline setting is parsed after taxonomy packages, so the default setting (online) is always set at this point. Switching the order that those two cli args are handled should resolve that that.

Reply:
Moved `internetConnectivity` option handling up.